data/locations/src/main/java/de/mm20/launcher2/locations/LocationSerialization.kt

```diff
@@ -0,0 +1,228 @@
+import de.mm20.launcher2.search.SavableSearchable
+import de.mm20.launcher2.search.SearchableDeserializer
+import de.mm20.launcher2.search.SearchableSerializer
+import de.mm20.launcher2.search.UpdateResult
+import de.mm20.launcher2.search.location.Address
+import de.mm20.launcher2.search.location.Attribution
+import de.mm20.launcher2.search.location.Departure
+import de.mm20.launcher2.search.location.LocationCategory
+import de.mm20.launcher2.search.location.OpeningHours
+import de.mm20.launcher2.search.location.OpeningSchedule
+import de.mm20.launcher2.serialization.Json
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import org.json.JSONArray
+import org.json.JSONObject
+import java.time.DayOfWeek
+import java.time.Duration
+import java.time.LocalTime
+
+@Serializable
+internal data class SerializedLocation(
+    val id: String? = null,
+    val lat: Double? = null,
+    val lon: Double? = null,
+    val category: LocationCategory? = null,
+    val label: String? = null,
+    val address: Address? = null,
+    val websiteUrl: String? = null,
+    val phoneNumber: String? = null,
+    val userRating: Float? = null,
+    val openingSchedule: OpeningSchedule? = null,
+    val timestamp: Long? = null,
+    val departures: List<Departure>? = null,
+    val fixMeUrl: String? = null,
+    val attribution: Attribution? = null,
+    val authority: String? = null,
+    val storageStrategy: StorageStrategy? = null,
+)
+
+internal fun getOpeningSchedule(json: JSONObject): OpeningSchedule {
+    fun getOpeningHours(array: JSONArray): ImmutableList<OpeningHours> {
+        val hours = mutableListOf<OpeningHours>()
+
+        for (i in 0 until array.length()) {
+            val json = array.getJSONObject(i)
+            val dayOfWeek =
+                DayOfWeek.of(json.optInt("day").takeIf { it in 1..7 } ?: continue)
+            val openingTimeMillis =
+                json.optLong("openingTime", -1).takeIf { it >= 0 } ?: continue
+            val durationMillis = json.optLong("duration", -1).takeIf { it >= 0 } ?: continue
+
+            hours.add(
+                OpeningHours(
+                    dayOfWeek = dayOfWeek,
+                    startTime = LocalTime.ofSecondOfDay(openingTimeMillis / 1000L),
+                    duration = Duration.ofMillis(durationMillis)
+                )
+            )
+        }
+
+        return hours.toPersistentList()
+    }
+
+    return if (json.optBoolean("isTwentyFourSeven")) {
+        OpeningSchedule.TwentyFourSeven
+    } else {
+        OpeningSchedule.Hours(
+            openingHours = json.optJSONArray("openingHours")?.let {
+                getOpeningHours(it)
+            } ?: persistentListOf()
+        )
+    }
+}
```

Comment:
Dead code, serialization happens in `:core:shared`, right?

Reply:
Yes, it's all handled by `kotlinx.serialization`